### DISCUSSION TITLE
Queue MessageCount from producer returning 0 when consumer attached to queue

### DISCUSSION THREAD

**Discussion by dapalex** [rabbitmq/rabbitmq-dotnet-client#1374] [General]
Hi,

I am having the same issue with RabbitMQ Client version 6.5.0 downloaded from nupkg.

My project is composed of a few containers which communicate each other through an event bus amqp-based (RabbitMQ).

The first container A is multithread working only as producer (it has a single shared connection among threads and one channel per thread). 
Each thread declares a durable queue using `IModel.QueueDeclare` method and send persistent messages.
The receiving container B (multiprocess) dequeues the messages from each queue (per process).

I downloaded the source code from github [here](https://github.com/rabbitmq/rabbitmq-dotnet-client/tree/v6.5.0) and ran quite a few tests.
While simply debugging with container A and the relating RabbitMQ container everything works fine: MessageCount returns the correct number of messages in the queue, running instead the whole project the issue occurs.
Playing with the containers I found out the issue occurs only when container B is subscribed to the queue as consumer: starting and stopping B, A consequently will show 0 messages when A is running and correct message count (thousands) when A stops.

Please let me know if further information is needed.

Thanks

_Originally posted by @dapalex in https://github.com/rabbitmq/rabbitmq-dotnet-client/discussions/1210#discussioncomment-6807801_

**Comment by michaelklishin** [comment]
`MessageCount` returns the number of messages in Ready state. If there is an active consumer in automatic acknowledgement mode or a with a high enough prefetch to consume everything, that value indeed will be zero, no matter how you retrieve it. Delivered but unacknowledged messages **will not be included**.

This value is returned by RabbitMQ and not computed by the client in any way.

**Comment by dapalex** [comment]
Hi,

Thanks for your prompt reply.

Actually in my case the consumption of messages is configured with manual Ack, and the number of messages is far higher than an eventual zeroing prefetch.

To give you a more concrete idea of what is going on I share some information:

- Screenshot of logs in container A

![Screenshot Container A](https://github.com/rabbitmq/rabbitmq-dotnet-client/assets/91396837/ff28759c-888d-4a53-bc79-240e68ab5feb)

- Screenshot of start logs in container B

![Screenshot Container B start](https://github.com/rabbitmq/rabbitmq-dotnet-client/assets/91396837/a95046db-832f-4553-a906-8b9d9c00a70e)

- Screenshot of stop logs in container B

![Screenshot Container B stop](https://github.com/rabbitmq/rabbitmq-dotnet-client/assets/91396837/cbeda5e0-ad01-4891-a281-89312ea3136f)

- Rabbimqctl output

![RMQCTL](https://github.com/rabbitmq/rabbitmq-dotnet-client/assets/91396837/f4a1aed8-609e-4e72-887f-38dddef1557f) output

- Screenshot of BasicConsume implementation

![BasicConsume](https://github.com/rabbitmq/rabbitmq-dotnet-client/assets/91396837/2dfb3691-85b7-4840-997e-f537df988de3)

- Tranche of logs for [A](https://github.com/rabbitmq/rabbitmq-dotnet-client/files/12431891/Logs.Container.A.txt)

- Tranche of logs for [B](https://github.com/rabbitmq/rabbitmq-dotnet-client/files/12431892/Logs.Container.B.txt)

Any hint on what could be wrong?

Thanks again

**Reply by lukebakken** [reply]
I can't really say much from a few screenshots. If you'd like us to investigate, please provide _code we can run_ to see this behavior.
